modules/channel-web/src/views/lite/index.tsx

```diff
@@ -1,30 +1,54 @@
 import { configure } from 'mobx'
-import { Provider } from 'mobx-react'
+import { Provider, observer, inject } from 'mobx-react'
 import DevTools from 'mobx-react-devtools'
 import React from 'react'
 import { IntlProvider } from 'react-intl'
 
 import Chat from './main'
 import { RootStore } from './store'
-import { availableLocale, defaultLocale, getUserLocale, initializeLocale, translations } from './translations'
+import { defaultLocale, translations } from './translations'
 configure({ enforceActions: 'observed' })
 
-export const Embedded = props => WebChat(props, false)
-export const Fullscreen = props => WebChat(props, true)
-
-initializeLocale()
-const locale = getUserLocale(availableLocale, defaultLocale)
-
-const WebChat = (props, fullscreen) => (
-  <IntlProvider locale={locale} messages={translations[locale]} defaultLocale={defaultLocale}>
-    <Provider store={new RootStore({ fullscreen })}>
-      <React.Fragment>
-        <Chat {...props} />
-        {process.env.NODE_ENV === 'development' && <DevTools className="bpw-mobx-tools" />}
-      </React.Fragment>
-    </Provider>
-  </IntlProvider>
-)
+export const Embedded = props => new Wrapper(props, false)
+export const Fullscreen = props => new Wrapper(props, true)
+
+interface State {
+  fullscreen: any
+  store: any
+}
+
+interface Props {}
+
+class ExposedWebChat extends React.Component<Props, State> {
```

Comment:
Please avoid creating new class when it's not required, especially for small components. Use react hooks for this. Also, make sure there are not too many re-rendering since it's quite heavy.

Reply:
It get re-render only on locale change. I haven't found a way to change locales without rerender of entire component, because of we could need to change it inside different places.
As for hooks - ok, thanks for advice.

Reply:
That's okay to re-render once when the language is changed, just need to double check to ensure it doesn't render more than that :)

Reply:
Yup, ty!